### ISSUE TITLE
🐛 Bug Report: Email from other targets/users are sent to all targets by default

### ISSUE THREAD

**Issue by JoshiJoshiJoshi** [appwrite/appwrite#8261]
### 👟 Reproduction steps

1. Add two or more users to a messaging topic.
2. Send an email either via console or api and select that topic.
OR
1. Send an email either via console or api and select two or more different users as target.

### 👍 Expected behavior

The users receives the email AND the email DOES NOT show the email from the other targets or users in the same topic.
Other users should not know the email from the other targets. This can be considered a data breach and maybe not GDPR compliant.

### 👎 Actual Behavior

By default Appwrite shows the user/target to all other targets as well.

### 🎲 Appwrite version

Version 1.5.x

### 💻 Operating system

Linux

### 🧱 Your Environment

Appwrite v1.5.7

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
Related discord discussion: https://discord.com/channels/564160730845151244/564160731327758347/1250438121733095444

**Comment by stnguyen90** [comment]
Another thread: https://discord.com/channels/564160730845151244/1298192741678710814

**Comment by alangautham** [comment]
![Image](https://github.com/user-attachments/assets/5ed1c04b-4fbb-4265-9157-0059c41ccf43)

Sent email to my app users (around 300) using topics from the appwrite console. Users received mail with other's mail ids exposed. Please refer to the attached screenshot. This is clearly a data breach and not GDPR compliant. Please fix it soon guys!

**Comment by 7oaker** [comment]
Hi, would be great if this gets on the bucket list for 2025 ;) 
Would save some trouble, also for data protection laws.

BR

**Comment by ChiragAgg5k** [comment]
@7oaker the team has started working on this, thanks your patience 🙏

**Comment by 7oaker** [comment]
Thanks a lot for your efforts @ChiragAgg5k 
i just tested but unfortunately it is the same:

![Image](https://github.com/user-attachments/assets/ce138fcc-035f-4892-8b18-bcae3e0429a3)

Beside that it would be nice to have a option to send an email to all users (i think you call it targets). Otherwise it can be very time consuming to go through all pages if you have lots of users.

The Message feature would be great, but without this fixes you cannot use it in praxis. :( 

Best Regards

**Comment by ChiragAgg5k** [comment]
@7oaker hi there, are you by chance using SMTP adapter for sending the emails? currently the fix only targets Mailgun and SMTP adapters.

also to send an email to multiple users, you should look into topics. they do exactly what you want, target users by subscribing them to specific groups- https://appwrite.io/docs/products/messaging/targets

**Comment by 7oaker** [comment]
@ChiragAgg5k yes i use SMTP Provider at the moment. Do you suggest swtiching or are fixes also planned for SMTP? 

Ah thanks a lot!

**Comment by ChiragAgg5k** [comment]
Closing — multi-recipient email privacy for Messaging is fixed across providers. Mailgun/Sendgrid use batch APIs, and SMTP no longer puts every target in a shared visible `To` list (recipients are sent via BCC).

Relevant changes:
- https://github.com/appwrite/appwrite/pull/9243
- https://github.com/appwrite/appwrite/pull/10644

Please reopen if you’re still seeing other recipients’ addresses exposed on a current release.
